src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -166,10 +174,6 @@ public function onSessionUsage(): void
             return;
         }
 
-        if (!$session = $this->container && $this->container->has('initialized_session') ? $this->container->get('initialized_session') : $requestStack->getCurrentRequest()->getSession()) {
```

Comment:
Are you 100% sure that moving this condition before the stateless check is safe ?